### ISSUE TITLE
NPE in :set when newVersion is the same as current version and using wildcards

### ISSUE THREAD

**Issue by jakub-bochenski** [mojohaus/versions-maven-plugin#38]
This should fail with some sensible message instead.

If e.g. current version is 0.3 and I run `mvn versions:set -DnewVersion=0.3  -DartifactId='*'` I get:

```
Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed.
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:108)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:116)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:361)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed.
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:144)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 19 more
Caused by: java.lang.NullPointerException
        at org.codehaus.mojo.versions.SetMojo.applyChange(SetMojo.java:276)
        at org.codehaus.mojo.versions.SetMojo.execute(SetMojo.java:245)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
```

**Comment by khmarbaise** [comment]
I have added an example project as integration test src/it/it-set-issue-88 on branch `issue-88` but i can't reproduce it. 
Do you have an working example of it?

**Comment by jakub-bochenski** [comment]
The issue occurs if you try to set the same version, in the test i see current version is `1.0` and new version is `2.0`

**Comment by andreasaronsson** [comment]
I get an NPE and a  stack trace that is very similar, if not identical with the command line `mvn versions:set -DnewVersion=x.x.x` with versions-maven-plugin-2.2. I.E. no asterisk or identical version. 

This error does not occur for me with versions-maven-plugin-2.1 nor with tip/2.3-SNAPSHOT/ddf38a13d20c111efbdf761a1f1c4e074ef9010f 

I think a release versions-maven-plugin-2.3 would be very much appreciated. :-)
Thanks for all your great work.

**Comment by prostagma** [comment]
fixed with 8ff41e560a667662938d27fe95a260c9f963f828
